contracts/src/LightClient.sol

```diff
@@ -118,7 +127,7 @@ contract LightClient {
         ) {
             revert OutdatedState();
         }
-        uint64 epochEndingBlockHeight = currentEpoch * BLOCKS_PER_EPOCH;
+        uint64 epochEndingBlockHeight = currentEpoch * blocksPerEpoch;
         bool isNewEpoch = finalizedState.blockHeight == epochEndingBlockHeight;
         if (!isNewEpoch && newState.blockHeight > epochEndingBlockHeight) {
             revert MissingLastBlockForCurrentEpoch(epochEndingBlockHeight);
```

Comment:
do you think we should package all these into a `if else` clause and skip it completely when there's no epoch change (i.e. when EPOCH_PER_BLOCK is max) ?

I'm hesitant. if we skip this in `else`, then users don't have to pay for the extra gas of "computing `isNewEpoch`" but our code looks slightly more complicated; OTOH, it's not too much gas, it's negligible compared to the verifier cost, so maybe it's perfectly fine for simpler logic sake.

wdyt?

Reply:
I think we can keep it as is. I added a TODO in 0c54374 so that we can come back to it when working on  gas optimization.